app/src/components/Admin/Institutions/Actions/AddInstitutionsButton.tsx

```diff
@@ -0,0 +1,151 @@
+import { useState } from "react";
+import GenericModalComponent from "@components/GenericModalComponent";
+
+import { Institution } from "@interfaces/Institution";
+import AuthServices from "@services/auth.services";
+
+import { useNotifications } from "@components/notification/NotificationContext";
+import { useApi } from "@hooks/useAPI";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+
+export const AddInstitutionButton = () => {
+  const [showModal, setShowModal] = useState(false);
+  const { addNotification } = useNotifications();
+  const navigate = useNavigate();
+  const {
+    call: addInstitution,
+    isLoading,
+    error,
+  } = useApi(AuthServices.addInstitution);
+
+  // Use-form setup
+  const { register, handleSubmit } = useForm<Institution>();
+
+  // Function to execute upon accepting an application
+  const onSubmit: SubmitHandler<Institution> = async (institutionData) => {
+    try {
+      const res = await addInstitution({
+        ...institutionData,
+      });
+      console.log(res);
+      setShowModal(false);
+      navigate("/educado-admin/applications");
+      addNotification(
+        "Adicionado " + res.institutionName + " como nova instituição"
+      );
+    } catch {
+      if (error) {
```

Comment:
Why have a check for error here? If we end up in the catch it is more than likely because of an error. Now i do not see an else statement here, that would mean that if this in not an error, we might not even know there is an error. Right? 

2 potential solutions: 
- Remove the check
- Reverse the if statement and add an "throw error"

I might also just be tripping 🥲 

>

Reply:
Good observation!
```
catch (error) {

}
```